integreat_cms/static/src/js/pois/poi-actions.ts

```diff
@@ -0,0 +1,93 @@
+import { getCsrfToken } from "../utils/csrf-token";
+
+window.addEventListener("load", () => {
+  document.getElementById("id_address")?.addEventListener("focusout", autoCompleteAddress);
+  document.getElementById("id_postcode")?.addEventListener("focusout", autoCompleteAddress);
+  document.getElementById("id_city")?.addEventListener("focusout", autoCompleteAddress);
+
+  let longitude = document.getElementById("id_longitude") as HTMLInputElement;
+  let latitude = document.getElementById("id_latitude") as HTMLInputElement;
+
+  document.getElementById("auto-fill-coordinates")?.addEventListener("input", ({ target }) => {
+    if ((target as HTMLInputElement).checked) {
+      autoCompleteAddress();
+    } else {
+      // Reset to initial value
+      longitude.value = longitude.dataset.initial;
+      latitude.value = latitude.dataset.initial;
+    }
+  });
+
+  // Add possibility to reset coordinates to initial values
+  if (longitude?.value) {
+    longitude.dataset.initial = longitude.value;
+  } else {
+    longitude.addEventListener("focusout", () => {
+      longitude.dataset.initial = longitude.value;
+    });
+  }
+  if (latitude?.value) {
+    latitude.dataset.initial = latitude.value;
+  } else {
+    latitude.addEventListener("focusout", () => {
+      latitude.dataset.initial = latitude.value;
+    });
+  }
+});
+
+async function autoCompleteAddress() {
+  let street = (document.getElementById("id_address") as HTMLInputElement).value;
+  let postcode = (document.getElementById("id_postcode") as HTMLInputElement).value;
+  let city = (document.getElementById("id_city") as HTMLInputElement).value;
+
+  // Only try auto filling if street and either postcode or city are given
+  if (street.trim().length == 0 || (postcode.trim().length == 0 && city.trim().length == 0)) {
+    return;
+  }
+
+  let autoFillCoordinates = document.getElementById("auto-fill-coordinates") as HTMLInputElement;
+  let error = document.getElementById("nominatim-error");
+  error.classList.add("hidden");
+
+  const response = await fetch(autoFillCoordinates.dataset.url, {
+    method: "POST",
+    headers: {
+      "X-CSRFToken": getCsrfToken(),
+    },
+    body: JSON.stringify({
+      street: street,
+      postcode: postcode,
+      city: city,
+    }),
+  });
+
+  const data = await response.json();
+
+  if (response.status != 200) {
+    error.textContent = data.error;
+    error.classList.remove("hidden");
+    return;
+  }
+
+  updateField("postcode", data.postcode);
+  updateField("city", data.city);
+  updateField("country", data.country);
```

Comment:
I moved the border animation into a separate function to reduce code duplication.